### DISCUSSION TITLE
Changement formulaire et formulaires en cours

### DISCUSSION THREAD

**Discussion by PhilippeLRETQC** [MTESSDev/FRW#3] [Annonces]
Bonjour,

Lorsque le fichier yml d'un formulaire change, quel est l'impact pour les clients qui ont des formulaires en cours selon le changement.

-Changement d'un libellé
- Retrait d'un champ
- Ajout d'un champ
- Changement de validation
- Autres cas pertinents à tenir en compte

Bonne journée

**Comment by anisite** [comment]
Bonjour,

- Changement d'un libellé
   Aucun impact
- Retrait d'un champ
   Aucun impact, la valeur est simplement non assignée et sera nettoyée lors de la transmission.
- Ajout d'un champ
   Si le champ est obligatoire l'utilisateur sera obligé de le renseigner pour soumettre
- Changement de validation
   Aucun impact connu 
- Autres cas pertinents à tenir en compte
   Normalement toutes les validations sont réexécutées, au moment d'écrire ce message, la seule chose très importante est le domaine de valeur d'un champ (radio, checkbox, select) si une option est retirée et qu'elle avait été au préalable sélectionnée elle sera soumise tel quel en sortie, le backend doit donc gérer durant une période des valeurs qui ne font plus parti des possibilitées, mais il y a un carte à ce sujet chez nous pour appliquer la vérification du domaine automatiquement plutot que par une règle manuelle de validation.

à ma connaissance il n'y a pas d'autres limitations, par contre si vous changez un libellé il pourrait être parfois logique de changer l'id de la question pour vous assurer que l'utilisateur ira la relire (advenant que la question change drastiquement la réponse à donner)

**Reply by anisite** [reply]
le dernier point avec la validation du domaine de valeur limité sera corrigée dans la prochaine version voir https://github.com/MTESSDev/FRW/issues/19#issuecomment-1439105734
